predictor/predictor.py

```
@@ -0,0 +1,347 @@
+            )
+            df["avg_runtime_minutes"] = cluster_metrics["avg_job_runtime"].fillna(0)
+
+            return df
+
+        except Exception as e:
+            print(f"Query error: {str(e)}")
+            raise
+
+    def predict(self):
+        try:
+            request_data = request.get_json()
```

Ruff (0.8.2)

Undefined name `request`

(F821)